tests/llmcompressor/transformers/compression/test_decompress.py

```diff
@@ -0,0 +1,133 @@
+            self.skeleton_model_stub,
+            torch_dtype=self.decompressed_model_hf_quantizer.dtype,
+            device_map=self.decompressed_model_hf_quantizer.device,
+        )
+
+        # decompression from HFQuantizer should populate weight_scale
+        assert hasattr(
+            self.decompressed_model_hf_quantizer.model.layers[0].self_attn.q_proj,
+            "weight_scale",
+        )
+
+        # dense model should not have weight_scale populated
+        assert not hasattr(
+            self.dense_model.model.layers[0].self_attn.q_proj, "weight_scale"
+        )
+
+        config = AutoConfig.from_pretrained(self.compressed_model_stub)
+
+        compression_config = getattr(config, QUANTIZATION_CONFIG_NAME, None)
+        self.compressor = ModelCompressor.from_compression_config(compression_config)
+        self.compressor.quantization_config.quantization_status = (
+            QuantizationStatus.FROZEN
+        )
+
+        # use the model_path to load the decompressed weights into dense_model
+        dense_model = copy.deepcopy(self.dense_model)
+
+        # overwrite the weights of the dense model
+        self.compressor.decompress(
+            model_path=self.compressed_model_stub,
+            model=self.dense_model,
+        )
+
+        # self.dense_model should be decompressed
+        assert dense_model is not self.dense_model
+
+        self.decompressed_model_manual = self.dense_model
+
+        assert hasattr(
+            self.decompressed_model_manual.model.layers[0].self_attn.q_proj,
+            "weight_scale",
+        )
+
+    def test_hf_quantizer_decompress_match_manual_decompress(self):
+        manual_device = self.decompressed_model_manual.device
+        decompressed_model_hf_quantizer = self.decompressed_model_hf_quantizer.device
+
+        self.decompressed_model_manual = self.decompressed_model_manual.to(
+            manual_device
+        )
+        self.decompressed_model_hf_quantizer = self.decompressed_model_hf_quantizer.to(
+            decompressed_model_hf_quantizer
+        )
+
+        for input in self.SAMPLE_INPUTS:
+            inputs = self.tokenizer(input, return_tensors="pt", padding=True).to(
+                self.decompressed_model_manual.device
+            )
+            inputs = inputs.to(self.decompressed_model_manual.device)
+
+            decompressed_model_manual_output = self.tokenizer.batch_decode(
+                self.decompressed_model_manual.generate(**inputs, max_length=50)
+            )
+
+            decompressed_model_hf_quantizer_out = self.tokenizer.batch_decode(
+                self.decompressed_model_hf_quantizer.generate(**inputs, max_length=50)
+            )
+
+            assert (
+                decompressed_model_hf_quantizer_out == decompressed_model_manual_output
+            )
+
+    @classmethod
+    def tearDownClass(self):
+        shutil.rmtree(self.test_dir)
+        del self.dense_model
```

Comment:
FYI there's no need to delete references manually. The references are dropped when the reference to the class is dropped, which should be when pytest finishes the test.